src/lib/ops.py

```diff
@@ -194,8 +194,9 @@ def U1(lam, d=1):
 def Rotation(v: np.ndarray, theta: float) -> np.ndarray:
   """Produce the single-qubit rotation operator."""
 
-  v = np.array(v)
-  if v.shape != (3,) or not np.isclose(a @ a, 1) or not np.all(np.isreal(v)):
+  v = np.asarray(v)
+  if (v.shape != (3,) or not np.isclose(a @ a, 1) or
```

Comment:
Same here. v @ v instead of a @ a? @xbe

Reply:
Fixed. The perils of not having automated testing.